distribution/lib/Standard/Base/0.0.0-dev/src/Data/Filter_Condition.enso

```diff
@@ -276,20 +276,20 @@ unify_condition_or_predicate : Filter_Condition | (Any -> Boolean) -> (Any -> Bo
 unify_condition_or_predicate (condition_or_predicate : Filter_Condition | Function) =
     case condition_or_predicate of
         condition : Filter_Condition -> condition.to_predicate
-        predicate -> handle_constructor_missing_arguments predicate
+        predicate -> handle_constructor_missing_arguments predicate predicate
 
 ## PRIVATE
 unify_condition_predicate_or_element condition =
     case condition of
         condition : Filter_Condition -> condition.to_predicate
-        predicate : Function -> handle_constructor_missing_arguments predicate
+        predicate : Function -> handle_constructor_missing_arguments predicate predicate
         element -> (== element)
 
 ## PRIVATE
    Checks if the given `function` is actually a `Filter_Condition` constructor
    that is just missing arguments. If so, it will report a more friendly error.
-   Otherwise it will just return the `function` as is.
-handle_constructor_missing_arguments function =
+   Otherwise it will run the `continuation`.
```

Comment:
This seems like something we'll need for other constructors, perhaps it should be made generic.

Reply:
Agree we may want to make this generic until we can improve type checking. Not one for this PR, but maybe next time we add one.

Reply:
The issue is - we are doing a bit of a heuristic to check if this is really a `Filter_Condition` - by looking at the text representation.

If we want to make it generic, we'd have to make this more robust as that does not seem like a good 'hack' to spread around.

Anyway, this only applies to situations where we have `X | Function` types. For types that do not accept a `Function` (most of our ADT args do not), it will be ideal to have this check done at runtime typecheck level.